plugins/redis/redis.cpp

```diff
@@ -0,0 +1,904 @@
+            return;
+        }
+    }
+    throwUnexpected();
+}
+void Connection::handleLockOnSet(ICodeContext * ctx, const char * key, const char * value, size_t size, unsigned expire)
+{
+    StringBuffer cmd("SET %b %b");
+    RedisPlugin::appendExpire(cmd, expire);
+
+    //Due to locking logic surfacing into ECL, any locking.set (such as this is) assumes that they own the lock and therefore go ahead and set regardless.
+    //It is possible for a process/call to 'own' a lock and store this info in the LockObject, however, this prevents sharing between clients.
+    redisAppendCommand(context, cmd.str(), key, strlen(key), value, size);//SET
+    StringBuffer channel;
+    encodeChannel(channel, key);
+    redisAppendCommand(context, "PUBLISH %b %b", channel.str(), channel.length(), value, size);//PUB
+
+    //Now read and assert replies
+    OwnedReply replyContainer = new Reply();
+    readReplyAndAssertWithKey(replyContainer, "SET", key);//SET reply
+    readReplyAndAssertWithKey(replyContainer, "PUB for the key", key);//PUB reply
+
+    //NOTE: Pipelining the above commands may not be the desired behaviour but instead only PUBLISH upon a successful SET. Doing both regardless, does however ensure
+    //(assuming only the SET fails) that any subscribers do in fact get their requested key-value even if the SET fails. However, this may not be expected behaviour
+    //as it's now possible for the key-value to actually exists in the cache when it was retrieved via redis plugin get function. This is documented in the README.
+    //Further more, it is possible that the locked value and thus the channel stored within the key is not that expected, i.e. computed via encodeChannel() (e.g.
+    //if set by a non-conforming external client/process). It is however, possible to account for this via using a GETSET instead of just the SET. This returns the old
+    //value stored, this can then be checked if it is a lock (i.e. has at least the "redis_key_lock prefix"), if it doesn't, PUB on the channel from encodeChannel(),
+    //otherwise PUB on the value retrieved from GETSET or possibly only if it at least has the prefix "redis_key_lock".
+    //This would however, prevent the two commands from being pipelined, as the GETSET would need to return before publishing.
+}
+//--------------------------------------------------------------------------------
+//                           ECL SERVICE ENTRYPOINTS
+//--------------------------------------------------------------------------------
+//-----------------------------------SET------------------------------------------
+ECL_REDIS_API void ECL_REDIS_CALL SyncLockRSetStr(ICodeContext * ctx, size32_t & returnSize, char * & returnValue, const char * key, size32_t valueSize, const char * value, const char * options, unsigned __int64 database, unsigned expire, const char * password, unsigned __int64 timeout)
```

Comment:
trivial: returnLength instead of returnSize would be consistent with the other functions.